common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -795,6 +796,20 @@ def _add_cache(self, course_version_guid, system):
         """
         if self.request_cache is not None:
             self.request_cache.data.setdefault('course_cache', {})[course_version_guid] = system
+            number_courses_in_cache = len(self.request_cache.data['course_cache'])
+            if number_courses_in_cache > 1:
+                # We shouldn't have any scenarios where there's more than
+                # one course in the request cache. My suspicion is that
+                # it would be indicative of a leak. So, if that happens,
+                # let's log it here with a traceback.
```

Comment:
This might not be true during re-runs, when you have both the original and the rerun of the course "active" in some sense during the same request.